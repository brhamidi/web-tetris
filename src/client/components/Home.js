import React, { useState } from 'react'
import { GameStatus } from '../actions'

const Home = ({ status, cb }) => {
	const [room, setRoom] = useState('');
	const [player, setPlayer] = useState('');

	const handleSubmit = (e) => {
		e.preventDefault();
		if (room !== '' && player !== '') {
			cb(room, player);
			window.location.hash = `#${room}[${player}]`;
		}
	}

	return (
		<div>
			{status}
			<h3> Home Page </h3>
			<form onSubmit={handleSubmit}>
				<label> room </label>
				<input
					type="text"
					value={room}
					onChange={(e) => setRoom(e.target.value)}
				/>
				<label> player </label>
				<input
					type="text"
					value={player}
					onChange={(e) => setPlayer(e.target.value)}
				/>
				<input type="submit" value="Submit" />
			</form>
		</div>
	)
}

export default Home;
