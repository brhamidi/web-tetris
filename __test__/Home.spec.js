import React from 'react';
import { shallow, mount} from 'enzyme';
import Home from '../src/client/components/Home';
import { List } from 'immutable';

describe('Home', () => {
	test('renders correctly', () => {
		const cb = jest.fn((a, b) => {});
		const onStart = jest.fn((a, b) => {});
		const wrapper = shallow(<Home OnStart={onStart} cb={cb} status={'ERROR'} gameList={List()} />);
		expect(wrapper).toMatchSnapshot();
	});
	test('Simulate wrong input value', () => {
		const cb = jest.fn((a, b) => {});
		const onStart = jest.fn((a, b) => {});
		const wrapper = mount(<Home OnStart={onStart} cb={cb} status={'ERROR'} gameList={List()} />);

		expect(cb.mock.calls.length).toBe(0);

		wrapper.find('form').simulate('submit');

		expect(cb.mock.calls.length).toBe(0);

		wrapper.find('input.room').simulate('change', {target:{value:"GOT"}});
		wrapper.find('input.player').simulate('change', {target:{value:"PDF"}});

		wrapper.find('form').simulate('submit');

		expect(cb.mock.calls.length).toBe(1);
	})
});
