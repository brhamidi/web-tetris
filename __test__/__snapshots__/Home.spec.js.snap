// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    className="status"
  />
  <h3
    className="title"
  >
    Home Page
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label>
      room
    </label>
    <input
      className="room"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      player
    </label>
    <input
      className="player"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
